src/Interpreters/Context.cpp

```diff
@@ -271,6 +272,7 @@ struct ContextSharedPart : boost::noncopyable
     mutable QueryCachePtr query_cache;         /// Cache of query results.
     mutable MMappedFileCachePtr mmap_cache; /// Cache of mmapped files to avoid frequent open/map/unmap/close and to reuse from several threads.
     ProcessList process_list;                               /// Executing queries at the moment.
+    SessionTracker session_tracker;
```

Comment:
The name is by the way quite generic and in a few weeks from now, nobody will be able to remember what this class is doing (except by opening the SessionTracker header and reading the code). What about something stupidly long but more expressive like `MaximumSessionsPerUserTracker maximum_sessions_per_user_tracker`?

Reply:
If the purpose of `SessionTracker `was just to check `Maximum session per user`, then yes, name would be more descriptive.
But because of possible futher plans to get info from `SessionTracker` about active sessions, it seems wrong to rename it right now to `MaximumSessionsPerUserTracker`.

Reply:
Alright, this is fine with me, of course.

(The reason I brought this up was because I read about the [YAGMI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) yesterday 😄 )